engine/client/voice.c

```diff
@@ -249,63 +510,274 @@ static uint Voice_GetOpusCompressedData( byte *out, uint maxsize, uint *frames )
 }
 
 /*
-===============================================================================
+=========================
+Voice_GetGSCompressedData
 
-	VOICE CHAT INTEGRATION
+Get compressed voice data for GoldSrc mode
+=========================
+*/
+static uint Voice_GetGSCompressedData( byte *out, uint maxsize, uint *frames )
+{
+    uint ofs = 0, size = 0;
+    const uint frame_size_samples = GS_DEFAULT_FRAME_SIZE;
+    const uint frame_size_bytes = GS_DEFAULT_FRAME_SIZE * voice.width;
+    static uint16_t sequence = 0;
+
+    if( voice.input_file )
+    {
+        uint numbytes;
+        double updateInterval, curtime = Sys_DoubleTime();
+
+        updateInterval = curtime - voice.start_time;
+        voice.start_time = curtime;
+
+        numbytes = updateInterval * voice.samplerate * voice.width * VOICE_PCM_CHANNELS;
+        numbytes = Q_min( numbytes, voice.input_file->size - voice.input_file_pos );
+        numbytes = Q_min( numbytes, sizeof( voice.input_buffer ) - voice.input_buffer_pos );
+
+        memcpy( voice.input_buffer + voice.input_buffer_pos, voice.input_file->buffer + voice.input_file_pos, numbytes );
+        voice.input_buffer_pos += numbytes;
+        voice.input_file_pos += numbytes;
+    }
+
+    if( !voice.input_file )
+        VoiceCapture_Lock( true );
+
+    *frames = 0;
+
+    while (voice.input_buffer_pos - ofs >= frame_size_bytes)
+    {
+        int bytes;
+
+        int is_silence = 1;
+        int16_t *samples = (int16_t *)(voice.input_buffer + ofs);
+        for (uint i = 0; i < frame_size_samples; ++i) {
+            if (samples[i] != 0) {
+                is_silence = 0;
+                break;
+            }
+        }
+        if (is_silence) {
+            ofs += frame_size_bytes;
+            continue;
+        }
+
+		if ( !voice.input_file )
+			Voice_ApplyGainAdjust(samples, frame_size_samples, voice_transmit_scale.value);
+
+        bytes = opus_encode( voice.gs_encoder, samples,
+            frame_size_samples, out + size + 4, maxsize - 4 );
```

Comment:
Same thing about magic numbers here.

If `4` represents 32-bit number, use `sizeof( uint32_t )`. If it represents two separate 16-bit values, use `sizeof( uint16_t ) * 2` to keep the code readable and maintainable.